server/embedded/src/org/labkey/embedded/ResourceType.java

```diff
@@ -0,0 +1,113 @@
+import static org.labkey.embedded.LabKeyServer.MAX_WAIT_MILLIS_DEFAULT;
+import static org.labkey.embedded.LabKeyServer.VALIDATION_QUERY_DEFAULT;
+
+public enum ResourceType
+{
+    jdbc
+            {
+                @Override
+                protected Map<String, String> mergeWithDefaults(Map<String, String> props)
+                {
+                    Map<String, String> result = super.mergeWithDefaults(props);
+                    result.putIfAbsent("type", DataSource.class.getName());
+                    result.putIfAbsent("maxTotal", MAX_TOTAL_CONNECTIONS_DEFAULT);
+                    result.putIfAbsent("maxIdle", MAX_IDLE_DEFAULT);
+                    result.putIfAbsent("maxWaitMillis", MAX_WAIT_MILLIS_DEFAULT);
+                    result.putIfAbsent("accessToUnderlyingConnectionAllowed", ACCESS_TO_CONNECTION_ALLOWED_DEFAULT);
+                    result.putIfAbsent("validationQuery", VALIDATION_QUERY_DEFAULT);
+                    return result;
+                }
+
+                @Override
+                public void addResource(String name, Map<String, String> props, StandardContext context) throws ConfigException
+                {
+                    String displayName = props.remove("displayName");
+                    if (displayName != null)
+                    {
+                        context.addParameter(name + ":DisplayName", displayName);
+                    }
+                    String logQueries = props.remove("logQueries");
+                    if (logQueries != null)
+                    {
+                        context.addParameter(name + ":LogQueries", displayName);
+                    }
+
+                    super.addResource(name, props, context);
+                }
+            },
+    jms
+            {
+                @Override
+                protected Map<String, String> mergeWithDefaults(Map<String, String> props)
+                {
+                    Map<String, String> result = super.mergeWithDefaults(props);
+                    result.putIfAbsent("type", "org.apache.activemq.ActiveMQConnectionFactory");
+                    result.putIfAbsent("factory", "org.apache.activemq.jndi.JNDIReferenceFactory");
+                    result.putIfAbsent("description", "JMS Connection Factory");
+                    result.putIfAbsent("brokerName", "LocalActiveMQBroker");
+                    return result;
+                }
+            },
+    ldap
+            {
+                @Override
+                protected Map<String, String> mergeWithDefaults(Map<String, String> props)
+                {
+                    Map<String, String> result = super.mergeWithDefaults(props);
+                    result.putIfAbsent("type", "org.labkey.premium.ldap.LdapConnectionConfigFactory");
+                    result.putIfAbsent("factory", "org.labkey.premium.ldap.LdapConnectionConfigFactory");
+                    return result;
+                }
+            },
```

Comment:
Since it is defined in the premium module, I would argue to remove the `ldap` resource type. This would also give some test coverage to the `generic` resource type by our existing ldap tests.

Reply:
OK, done. From a quick search all configs, including Readme files, include explicit values for those two properties